### PR TITLE
ENH, MAINT: voronoi_plot_2d nicer inf lines

### LINKED ISSUE #19653: ENH: Voronoi diagram gives unexpected results from scipy.spatial
### Describe your issue.

When I try to plot the Voronoi diagram using the scipy.spatial, the results show (please see in the reproducing code example section), insstead of getting 5 areas, I am getting 4, even though I provide 5 points in the 2D space. Moreover the space separation seems to be a bit off and they are all rectangles

### Reproducing Code Example

```python
import pandas as pd
pd.DataFrame({'cl': {0: 'A', 1: 'C', 2: 'H', 3: 'M', 4: 'S'},
 'd': {0: 245.059986986012,
  1: 320.49044143557785,
  2: 239.79023081978914,
  3: 263.38325791238833,
  4: 219.53334398353175},
 'p': {0: 10.971011721360075,
  1: 10.970258360366753,
  2: 13.108487516946218,
  3: 12.93241352743668,
  4: 13.346107628161008}})

import numpy as np
import matplotlib.pyplot as plt
from scipy.spatial import Voronoi, voronoi_plot_2d

centers2 = np.array(
    centers_dt[['d', 'p']]
)

scatter_x = np.array(centers_dt['d'])
scatter_y = np.array(centers_dt['p'])
group = np.array(centers_dt['cl'])
cdict = {'C': 'red', 'A': 'blue', 'H': 'green', 'M': 'yellow', 'S': 'black'}

fig, ax = plt.subplots()
for g in np.unique(group):
    ix = np.where(group == g)
    ax.scatter(scatter_x[ix], scatter_y[ix], c = cdict[g], label = g, s = 100)
ax.legend()

vor = Voronoi(centers2)
fig = voronoi_plot_2d(vor,plt.gca())

plt.show()
plt.close()
```

### Error message

```shell
There is no error message, it is just that the results are unexpected.
```

### SciPy/NumPy/Python version and system information

```shell
1.7.0 1.21.6 sys.version_info(major=3, minor=7, micro=10, releaselevel='final', serial=0)
lapack_mkl_info:
  NOT AVAILABLE
openblas_lapack_info:
    libraries = ['openblas', 'openblas']
    library_dirs = ['/usr/local/lib']
    language = c
    define_macros = [('HAVE_CBLAS', None)]
lapack_opt_info:
    libraries = ['openblas', 'openblas']
    library_dirs = ['/usr/local/lib']
    language = c
    define_macros = [('HAVE_CBLAS', None)]
blas_mkl_info:
  NOT AVAILABLE
blis_info:
  NOT AVAILABLE
openblas_info:
    libraries = ['openblas', 'openblas']
    library_dirs = ['/usr/local/lib']
    language = c
    define_macros = [('HAVE_CBLAS', None)]
blas_opt_info:
    libraries = ['openblas', 'openblas']
    library_dirs = ['/usr/local/lib']
    language = c
    define_macros = [('HAVE_CBLAS', None)]
```

### PR DESCRIPTION
**Pull request by tylerjereddy**
* Fixes #19653

* we currently set the plotted length of infinitely-long Voronoi edges as a function of the largest x-or-y Voronoi generator range; in cases where the range of one axis is substantially larger than the other, it seems that we can produce a more appealing representation of the infinitely-long line by accounting for the aspect ratio; otherwise, there's a danger the (infinitely-long) Voronoi edge may not be visible with our default `voronoi_plot_2d` settings

* note that we only have very crude testing for `voronoi_plot_2d`, so there is no guarantee this doesn't introduce new problems for other scenarios (so not keen on backporting, even if folks thought this was a genuine bug), and I'm also not super keen on extracting line data from matplotlib like this, but I do also think this probably is a step in the right testing direction if we're going to respond to user feedback in a sane way; the function already takes 8 parameters, and to some extent the user should be expected to make reasonable adjustments, but since the Voronoi edge was invisible in this default case I thought we might try to fix it

* while in the neighborhood, remove some local test shims for `matplotlib` `1.x`; `matplotlib` is already at `3.x` so I think we can drop those

[skip cirrus]

cc @bagrow and @dpapadop for feedback, since they were vocal on the original ticket

Before/after for the generators in question (it may be good to spot check a few more cases and perhaps even suggest another regression case or two):

image

image

### CODE REVIEW

File: scipy/spatial/tests/test__plotutils.py [hunk 1]
```diff
@@ -46,9 +42,29 @@ def test_convex_hull(self):
...
+        ax = fig.gca()
+        infinite_segments = ax.collections[1].get_segments()
+        expected_segments = np.array([[[282.77256, -254.76904],
+                                      [282.729714, -4544.744698]],
+                                     [[282.77256014, -254.76904029],
+                                      [430.08561382, 4032.67658742]],
+                                     [[229.26733285,  -20.39957514],
+                                      [-168.17167404, -4291.92545966]],
+                                     [[289.93433364, 5151.40412217],
+                                      [330.40553385, 9441.18887532]]])
```
Comment (lucascolley) [inline scipy/spatial/tests/test__plotutils.py:69]:
```suggestion
        expected_segments = np.array([[[282.77256, -254.76904],
                                       [282.729714, -4544.744698]],
                                      [[282.77256014, -254.76904029],
                                       [430.08561382, 4032.67658742]],
                                      [[229.26733285,  -20.39957514],
                                       [-168.17167404, -4291.92545966]],
                                      [[289.93433364, 5151.40412217],
                                       [330.40553385, 9441.18887532]]])
```

File: scipy/spatial/tests/test__plotutils.py [hunk 2]
```diff
@@ -46,9 +42,29 @@ def test_convex_hull(self):
...
+
+    def test_gh_19653(self):
+        # aspect ratio sensitivity of voronoi_plot_2d
+        # infinite Voronoi edges
+        points = np.array([[245.059986986012, 10.971011721360075],
+                           [320.49044143557785, 10.970258360366753],
+                           [239.79023081978914, 13.108487516946218],
+                           [263.38325791238833, 12.93241352743668],
+                           [219.53334398353175, 13.346107628161008]
+                          ])
```
Comment (lucascolley) [inline scipy/spatial/tests/test__plotutils.py:57]:
```suggestion
        points = np.array([[245.059986986012, 10.971011721360075],
                           [320.49044143557785, 10.970258360366753],
                           [239.79023081978914, 13.108487516946218],
                           [263.38325791238833, 12.93241352743668],
                           [219.53334398353175, 13.346107628161008]])
```

**Review by lucascolley — APPROVED** [review]
I can't speak to the `aspect_factor` change but the test looks good so happy if CI is happy.

Would be nice to keep the style consistent at least locally here hence nits inline.

**Comment by tylerjereddy** [comment]
I tried to address the reviewer-suggested formatting changes. I also added another regression test, this time for a less extreme aspect ratio.

**Comment by ev-br** [comment]
Core dev approved, new tests look thorough to  a non-expert, modifications to old tests look good. Merged, thanks Tyler, Lucas

**Pull request merged**